File: src/main/java/de/iteratec/iteraOfficeMap/workplace/WorkplacesController.java
package de.iteratec.iteraOfficeMap.workplace;

import io.swagger.annotations.ApiOperation;
import io.swagger.annotations.ApiResponse;
import io.swagger.annotations.ApiResponses;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

import java.util.List;

import static java.util.stream.Collectors.toList;

@RestController
@RequestMapping("/api")
public class WorkplacesController {

    private final WorkplaceService workplaceService;

    @Autowired
    public WorkplacesController(WorkplaceService workplaceService) {
        this.workplaceService = workplaceService;
    }

    @ApiOperation(value = "getAllWorkplaces", notes = "Returns a list of all workplaces.")
    @RequestMapping(method = RequestMethod.GET, value = "/allworkplaces")
    public List<WorkplaceDTO> allWorkplaces() {
        return workplaceService.getAllWorkplaces().stream().map(WorkplaceDTO::new).collect(toList());
    }

    @ApiOperation(value = "getStatus", notes = "Returns a list of all workplaces.")
    @RequestMapping(method = RequestMethod.GET, value = "/getstatus")
    public String getStatus(Long workplaceID) {
        return workplaceService.getStatus(workplaceID).toString();
    }

    @ApiOperation(value = "addNewWorkplace", notes = "Adds one new workplace to the database.")
    @RequestMapping(method = RequestMethod.POST, value = "/addworkplace", consumes = MediaType.APPLICATION_JSON_VALUE)
    @ResponseStatus(HttpStatus.OK)
    @ApiResponses(value = {@ApiResponse(code = 406, message = "Entry already exists")})
    public void addWorkplace(@RequestBody AddWorkplaceDTO workplaceDTO) {
        workplaceService.addNewWorkplace(new Workplace(workplaceDTO.getName(), workplaceDTO.getX(), workplaceDTO.getY(), workplaceDTO.getMapId(), workplaceDTO.getEquipment()));
    }

    @ApiOperation(value = "deleteWorkplace", notes = "Deletes one workplace from the database.")
    @RequestMapping(method = RequestMethod.DELETE, value = "/deleteworkplace", consumes = MediaType.APPLICATION_JSON_VALUE)
    @ApiResponses(value = {@ApiResponse(code = 406, message = "Entry does not exists")})
    public void deleteWorkplace(@RequestBody DeleteWorkplaceDTO deleteWorkplaceDTO) {
        workplaceService.deleteWorkplace(deleteWorkplaceDTO.getId());
    }

    @ApiOperation(value = "updateWorkplace", notes = "Updates one workplace in the database.")
    @RequestMapping(method = RequestMethod.PUT, value = "/updateworkplace", consumes = MediaType.APPLICATION_JSON_VALUE)
    @ApiResponses(value = {@ApiResponse(code = 406, message = "Entry does not exists")})
    public void updateWorkplace(@RequestBody UpdateWorkplaceDTO workplaceDTO) {
        Workplace updateWorkplace = new Workplace(workplaceDTO.getName(), workplaceDTO.getX(), workplaceDTO.getY(), workplaceDTO.getMapId(), workplaceDTO.getEquipment());
        updateWorkplace.setId(workplaceDTO.getId());
        workplaceService.updateWorkplace(updateWorkplace);
    }

}

File: src/main/java/de/iteratec/iteraOfficeMap/workplace/WorkplacesController.kt
package de.iteratec.iteraOfficeMap.workplace

import io.swagger.annotations.ApiOperation
import io.swagger.annotations.ApiResponse
import io.swagger.annotations.ApiResponses
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api")
class WorkplacesController @Autowired constructor(private val workplaceService: WorkplaceService) {

    @GetMapping("/allworkplaces")
    @ApiOperation(value = "getAllWorkplaces", notes = "Returns a list of all workplaces.")
    fun allWorkplaces(): List<WorkplaceDTO> {
        return workplaceService.getAllWorkplaces()
                .map { WorkplaceDTO(it) }
    }

    @GetMapping("/getstatus")
    @ApiOperation(value = "getStatus", notes = "Returns a list of all workplaces.")
    fun getStatus(workplaceID: Long?): String {
        return workplaceService.getStatus(workplaceID!!).toString()
    }

    @PostMapping("/addworkplace", consumes = [MediaType.APPLICATION_JSON_VALUE])
    @ApiOperation(value = "addNewWorkplace", notes = "Adds one new workplace to the database.")
    @ResponseStatus(HttpStatus.OK)
    @ApiResponses(value = [ApiResponse(code = 406, message = "Entry already exists")])
    fun addWorkplace(@RequestBody workplaceDTO: AddWorkplaceDTO) {
        workplaceService.addNewWorkplace(Workplace(null, workplaceDTO.name, workplaceDTO.x, workplaceDTO.y, workplaceDTO.mapId, workplaceDTO.equipment))
    }

    @DeleteMapping("/deleteworkplace", consumes = [MediaType.APPLICATION_JSON_VALUE])
    @ApiOperation(value = "deleteWorkplace", notes = "Deletes one workplace from the database.")
    @ApiResponses(value = [ApiResponse(code = 406, message = "Entry does not exists")])
    fun deleteWorkplace(@RequestBody deleteWorkplaceDTO: DeleteWorkplaceDTO) {
        workplaceService.deleteWorkplace(deleteWorkplaceDTO.id)
    }

    @PutMapping("/updateworkplace", consumes = [MediaType.APPLICATION_JSON_VALUE])
    @ApiOperation(value = "updateWorkplace", notes = "Updates one workplace in the database.")
    @ApiResponses(value = [ApiResponse(code = 406, message = "Entry does not exists")])
    fun updateWorkplace(@RequestBody workplaceDTO: UpdateWorkplaceDTO) {
        val updateWorkplace = Workplace(workplaceDTO.id, workplaceDTO.name, workplaceDTO.x, workplaceDTO.y, workplaceDTO.mapId, workplaceDTO.equipment)
        workplaceService.updateWorkplace(updateWorkplace)
    }

}
